opengovsg/askgovsg
# refactor(spinner): migrate to Chakra and TypeScript

- Convert Spinner component to use Chakra components and TypeScript
- AnswerSection component uses Chakra's SkeletonText directly

## Problem

Closes #273 

## Solution

Previously, 2 types of spinners were being used to display the loading state - one for pages and another for non-page sections (e.g. answer section, sending OTP)
These have been replaced with the Chakra spinner. For the answer section, Chakra's SkeletonText is used directly.


## Before & After Screenshots

**BEFORE**:

https://user-images.githubusercontent.com/56983748/133585462-4a283630-2281-4c9e-97e6-ce011d83c8aa.mov


https://user-images.githubusercontent.com/56983748/133585485-b1f53a1c-c96e-4c29-b612-508c72848352.mov




**AFTER**:



https://user-images.githubusercontent.com/56983748/133585788-20bd8abd-ff0d-4dd5-9727-63f169ed800d.mov



https://user-images.githubusercontent.com/56983748/133585621-fb1f08ea-fdca-48dc-a8a2-a7a684cbbe66.mov


<img src="https://user-images.githubusercontent.com/56983748/133721691-684d24e7-35b9-45ce-82c3-1a8a85167206.jpg" height="500px">

<img src="https://user-images.githubusercontent.com/56983748/133722268-5d552904-c740-4fe5-bc9a-900bb427c93b.jpg" height="500px">


## Tests

- Verify that pages use the updated Chakra spinners. These include
  - Landing page
  - Answer page
  - Header
  - Login page (sending OTP)
  - Edit form page
  - Post form page
  - Search results page
